code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -2021,6 +2033,63 @@
 	taste_description = "bananas"
 	can_synth = TRUE
 
+/datum/reagent/wittel
+	name = "Wittel"
+	description = "An extremely rare metallic-white substance only found on demon-class planets."
+	color = "#FFFFFF" // rgb: 255, 255, 255
+	taste_mult = 0 // oderless and tasteless
+
+/datum/reagent/metalgen
+	name = "Metalgen"
+	data = list("material"=null)
+	description = "A purple metal morphic liquid, said to impose it's metallic properties on whatever it touches."
+	color = "#b000aa"
+	taste_mult = 0 // oderless and tasteless
+	var/applied_material_flags = MATERIAL_ADD_PREFIX | MATERIAL_COLOR
+	var/minumum_material_amount = 100
+
+/datum/reagent/metalgen/reaction_obj(obj/O, volume)
+	metal_morph(O)
+	return
+
+/datum/reagent/metalgen/reaction_turf(turf/T, volume)
+	metal_morph(T)
+	return
+
+/datum/reagent/metalgen/proc/metal_morph(atom/A)
+	var/metal_dat = data["material"]
+	if(!metal_dat)
+		return
+	var/metal_amount = 0
+
+	for(var/B in A.custom_materials)
+		metal_amount += A.custom_materials[B]
+
+	if(!metal_amount)
+		metal_amount = minumum_material_amount
+
+	A.material_flags = applied_material_flags
+	A.set_custom_materials(list(getmaterialref(metal_dat) = metal_amount))
```

Comment:
You should have the instance before you even get to this proc, replace the type at some point during creation of the reagent